torch/ao/quantization/fx/prepare.py

```diff
@@ -239,20 +239,91 @@ def is_pattern_dtype_config_supported_by_backend(
         supported = True
-        for arg in input_node.args:
-            supported = supported and \
-                is_input_arg_dtype_supported_by_backend(
-                    arg, input_node, node_name_to_target_dtype, dtype_config, backend_config)
-        for k, arg in input_node.kwargs.items():
-            supported = supported and \
-                is_input_arg_dtype_supported_by_backend(
-                    arg, input_node, node_name_to_target_dtype, dtype_config, backend_config)
+        for arg in list(input_node.args) + list(input_node.kwargs.values()):
+            supported = supported and is_input_arg_dtype_supported_by_backend(
+                arg, input_node, node_name_to_target_dtype, dtype_config, backend_config)
         # check if output dtype is supported
         supported = supported and is_output_dtype_supported_by_backend(
             output_node, node_name_to_target_dtype, dtype_config)
         if supported:
-            return True
-    return False
+            return dtype_config
+    return None
+
+def _get_qconfig_compatible_with_dtype_config(
+        qconfig: QConfig,
+        dtype_config: DTypeConfig,
+        pattern: Pattern) -> QConfig:
+    """
+    Return an updated QConfig that satisfies constraints from the backend specified through `dtype_config`.
+
+    We require the QConfig quantization range to be a subset of the backend's quantization range,
+    and the QConfig min scale value to be equal to the backend's min scale value. If either the
+    quantization range or the min scale value is not specified in the QConfig, then we will borrow
+    the corresponding values from the backend.
+
+    If no changes to the QConfig are required, then the original QConfig is returned as is.
+    """
+    def get_new_activation_post_process(
+            activation_post_process: Union[_PartialWrapper, Type[ObserverBase]],
+            dtype_with_constraints: DTypeWithConstraints,
+            debug_string: str) -> Union[_PartialWrapper, Type[ObserverBase]]:
+        """
+        Return an updated activation post process that satisfies the constraints of the backend.
+        If the backend constraints cannot be satisfied, an error will be thrown instead.
+        If no changes to the activation post process are required, then the original activation
+        post process is returned as is.
+        """
+        extra_keywords = {}
+        backend_quant_min = dtype_with_constraints.quant_min
+        backend_quant_max = dtype_with_constraints.quant_max
+        backend_scale_min = dtype_with_constraints.scale_min
+        if isinstance(activation_post_process, _PartialWrapper):
+            qconfig_quant_min = activation_post_process.p.keywords.get("quant_min")
+            qconfig_quant_max = activation_post_process.p.keywords.get("quant_max")
+            # TODO: rename eps to scale_min
+            qconfig_scale_min = activation_post_process.p.keywords.get("eps")
```

Comment:
Not sure if these are equivalent, I'd expect something like `scale_min = max(scale_min, eps)`, where `scale_min` is positive.

Reply:
OK, we can require `scale_min >= eps` instead

Reply:
`scale_min` is the scale calculated by the user (through observation, hardcoded, etc).  `eps` is a term to prevent division by zero, the reason `eps` is there is to not have a restriction on `scale_min`.  If I remember the restriction correctly, it's on `eps` and not on `scale_min`.

Reply:
we set `eps` in the qnnpack symmetric qconfig previously because there is no `scale_min`. I think `eps` has the same effect as `scale_min`, and `scale_min` would make more sense if we also have `scale_max` so that we have `scale_min` and `scale_max` instead of `eps` and `scale_max`